tool_packages/unique_internal_search/unique_internal_search/service.py

```diff
@@ -299,6 +327,82 @@ def _get_max_tokens(self) -> int:
             )
             return self.config.max_tokens_for_sources
 
+    async def _create_or_update_active_message_log(
+        self,
+        *,
+        progress_message: str | None = None,
+        chunks: list[ContentChunk] | None = None,
+        search_strings_list: list[str],
+        status: MessageLogStatus,
+    ) -> MessageLog | None:
+        if self._message_step_logger is None:
+            return None
+        if chunks is not None:
+            message_log_reference_list = (
+                await self._define_reference_list_for_message_log(
+                    content_chunks=chunks,
+                )
+            )
+        else:
+            message_log_reference_list = []
+        details = await self._prepare_message_log_details(
+            query_list=search_strings_list
+        )
+        return self._message_step_logger.create_or_update_message_log(
+            active_message_log=self._active_message_log,
+            header=self._display_name,
+            progress_message=progress_message,
+            status=status,
+            details=details,
+            references=message_log_reference_list,
+        )
+
+    async def _define_reference_list_for_message_log(
+        self,
+        *,
+        content_chunks: list[ContentChunk],
+    ) -> list[ContentReference]:
+        """
+        Create a reference list for internal search content chunks.
+
+        Args:
+            content_chunks: List of ContentChunk objects to convert
+        Returns:
+            List of ContentReference objects
+        """
+        data: list[ContentReference] = []
+        count = 0
+        for content_chunk in content_chunks:
```

Comment:
This is a good place to use the `enumerate` function, that way you can get rid of the `count` variable